src/cli/commands/fellowship/propose.ts

```diff
@@ -0,0 +1,176 @@
+import { Command } from "commander";
+import chalk from "chalk";
+import ora from "ora";
+import { createPublic, createWallet } from "../../client";
+import { resolvePk } from "../../getPk";
+import GameMaster from "../../../rankify/GameMaster";
+import { CLIUtils } from "../../utils";
+import InstanceBase from "../../../rankify/InstanceBase";
+import RankifyPlayer from "../../../rankify/Player";
+import { Address, Hex, bytesToHex, hexToBytes } from "viem";
+import * as secp256k1 from "@noble/secp256k1";
+import { gameStatusEnum } from "../../../types";
+
+export const propose = new Command("propose")
+  .description("Submit a proposal for a game turn")
+  .argument("<instance>", "Address or index of the Rankify instance")
+  .argument("<gameId>", "ID of the game to propose for")
+  .option("-r, --rpc <url>", "RPC endpoint URL. If not provided, RPC_URL environment variable will be used")
+  .option(
+    "-k, --key <privateKey>",
+    "Private key or index to derive from mnemonic for signing transactions. If not provided, PRIVATE_KEY environment variable will be used"
+  )
+  .option(
+    "-t, --title <title>",
+    "Title of the proposal", "Default title"
+  )
+  .option(
+    "-b, --body <body>",
+    "Body content of the proposal", "Default body"
+  )
+  .option(
+    "-f, --file <filePath>",
+    "Path to a markdown file containing the proposal content"
+  )
+  .action(async (instanceAddress, gameId, options) => {
+    const spinner = ora("Initializing clients...").start();
+
+    try {
+      // Initialize clients
+      const publicClient = await createPublic(options.rpc);
+      const privateKey = resolvePk(options.key, spinner);
+      const walletClient = await createWallet(options.rpc, privateKey);
+      const chainId = Number(await publicClient.getChainId());
+      const account = walletClient.account?.address;
+
+      if (!account) {
+        spinner.fail("No account available");
+        throw new Error("No account available");
+      }
+
+      // Resolve instance address
+      const resolvedInstanceAddress = await CLIUtils.resolveInstanceAddress(
+        instanceAddress,
+        chainId,
+        publicClient,
+        spinner
+      );
+
+      // Create game master client
+      spinner.text = "Creating game master client...";
+      const gmWalletClient = await createWallet(options.rpc);
+      
+      const gameMaster = new GameMaster({
```

Comment:
Either signature shall be passed as argument, or GM key. 

It makes more sense to pass signature as arument to make this CLI client-side usable..